src/FiltrationGroupComponent/types.ts

```diff
@@ -0,0 +1,20 @@
+type GroupType = "and" | "or";
+
+type ConditionState = "view" | "edit" | "shaded";
+
+export interface StateProps {
+	groupType: GroupType; // тип группы: И или ИЛИ
+	state: ConditionState;
+	andLabel: string; // лейбл для И
+	orLabel: string; // лейбл для ИЛИ
+	shouldShowLabel?: boolean; // нужно ли отображать лейбл на брекете группы
+	children: React.ReactNode[]; // условия фильтрации внутри группы. могут быть FiltrationGroupComponent или FiltrationConditionComponent
```

Comment:
Всё-таки больше подходит так:
```suggestion
	children?: React.ReactNode | React.ReactNode[]; // условия фильтрации внутри группы. могут быть FiltrationGroupComponent или FiltrationConditionComponent
```